packages/core/echo/echo-db/src/core-db/core-database-query-context.ts

```diff
@@ -163,25 +164,53 @@ export class CoreDatabaseQueryContext implements QueryContext {
     }
   }
 
-  private _filterMapCore(
+  private async _filterMapCore(
     filter: Filter,
     core: ObjectCore,
     queryStartTimestamp: number,
     result: RemoteQueryResult | undefined,
-  ): QueryResult | null {
+  ): Promise<QueryResult | null> {
     if (!filterMatch(filter, core)) {
       return null;
     }
 
+    const data = await this._recursivelyJoinFields(core.toPlainObject(), filter.options.include);
+
     return {
       id: core.id,
       spaceId: core.database!.spaceId,
       spaceKey: core.database!.spaceKey,
-      object: core.toPlainObject(),
+      object: data,
       match: result && { rank: result.rank },
       resolution: { source: 'remote', time: Date.now() - queryStartTimestamp },
     } satisfies QueryResult;
   }
+
+  private async _recursivelyJoinFields(
```

Comment:
todo: handle recursive references?
```ts
const object = { parent: null, child: {} };
object.child.parent = object;
```

Reply:
This will require the JoinSpec object to be recursive which we would wanna prohibit